### ISSUE TITLE
task inputs to be passed from top level task to sub-task using --with cli flag

### ISSUE THREAD

**Issue by MxNxPx** [defenseunicorns/uds-cli#414]
### Is your feature request related to a problem? Please describe

I cannot seem to figure out a way to provide [an input](https://github.com/defenseunicorns/uds-cli/blob/main/docs/runner.md#task-inputs-and-reusable-tasks) to a top level task that is passed to an imported sub-task when using the cli `--with` flag, so am assuming that is a new feature request.

### Describe the solution you'd like

- **Given** a task is defined in my top-level ./tasks.yaml that references a sub-task via import
- **When** calling this subtask vi cli and passing in an input using the `--with` flag
- **Then** the input is passed through and leveraged in the subtask

### Additional context

Here are the steps I took to test whether the functionality works today...

```console
## create subtask dir and subtask
# mkdir tasks
# cat <<EOF>tasks/echo.yaml
tasks:
  - name: echo-var
    inputs:
      hello-input:
        default: hello world
        description: This is an input to the echo-var task
      deprecated-input:
        default: foo
        description: this is a input from a previous version of this task
        deprecatedMessage: this input is deprecated, use hello-input instead
    actions:
      # to use the input, reference it using INPUT_<INPUT_NAME> in all caps
      - cmd: echo \$INPUT_HELLO_INPUT

  - name: use-echo-var
    actions:
      - task: echo-var
        with:
          # hello-input is the name of the input in the echo-var task, hello-unicorn is the value we want to pass in
          hello-input: hello unicorn
EOF

## create top level task that imports subtask
# cat <<EOF>tasks.yaml
> includes:
  - echo: ./tasks/echo.yaml

tasks:
  - name: default
    actions:
      - task: echo:echo-var
EOF

## run the cli command passing in the input using the `--with` flag
# uds run --with hello-input="hello there"

 NOTE  Saving log file to /tmp/uds-2024-02-09-09-58-41-1322960881.log

 WARNING  Task default does not have an input named hello-input
     hello world                                                                                                                                                                                                                                                                                                      
  ✔  Completed "echo $INPUT_HELLO_INPUT"
  
## this does work when specifying the sub-task file via cli
# uds run echo-var -f tasks/echo.yaml --with hello-input="hello there"

 NOTE  Saving log file to /tmp/uds-2024-02-09-09-59-35-3699030165.log
     hello there                                                                                                                                                                                                                                                                                                      
  ✔  Completed "echo $INPUT_HELLO_INPUT"   
```

**Comment by UncleGedd** [comment]
Hey @MxNxPx we are actually planning on deprecating `--with` because of all the confusion it's causing. It'll be replaced with `--set- and the following syntax:
```yaml
variables:
  - name: FOO
    default: your-mom

tasks:
  - name: echo-var
    inputs:
      hello-input:
        default: hello world
        description: This is an input to the echo-var task
    actions:
      - cmd: echo hello -- $INPUT_HELLO_INPUT

  - name: use-echo-var
    actions:
      - task: echo-var
        with:
          hello-input: ${FOO}
```

You will be able to pass the variable `FOO` into `with.hello-input` via:
- the variable's default
- an env var prefixed with `UDS_` (so `UDS_FOO`)
- using `--set foo=bar`

Thoughts on this approach?

**Comment by UncleGedd** [comment]
Closed #415
